test/acceptance/tests/terminating-gateway/terminating_gateway_test.go

```diff
@@ -0,0 +1,94 @@
+package connect
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that terminating gateways work in a default installation.
+func TestTerminatingGateway(t *testing.T) {
+	testName := "Terminating GW non-acl"
+	t.Run(testName, func(t *testing.T) {
+		env := suite.Environment()
+
+		helmValues := map[string]string{
+			"connectInject.enabled":                    "true",
+			"terminatingGateways.enabled":              "true",
+			"terminatingGateways.gateways[0].name":     "terminating-gateway",
+			"terminatingGateways.gateways[0].replicas": "1",
+		}
+
+		t.Log("creating consul cluster")
+		releaseName := helpers.RandomName()
+		consulCluster := framework.NewHelmCluster(t, helmValues, env.DefaultContext(t), suite.Config(), releaseName)
+		consulCluster.Create(t)
+
+		// Once the cluster is up register the external service, then create the config entry.
+		consulClient := consulCluster.SetupConsulClient(t, false)
+
+		// Register the external service
+		t.Log("registering the service")
+		if err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+			Address:           "example.com",
```

Comment:
Should we instead create a static-server pod (non connect injected) and a kube service and use that instead? Wondering if this will be flaky due to depending on an external website (example.com).

Reply:
@ishustava do you have an opinion on this?

Reply:
🤔 Looking at the docs, it seems that the primary purpose of this is documentation. This means that we'd be abusing this in our tests, even if it's not going to cause flakiness. I do like that we're using an actual external-to-k8s service rather than a static server. Is there a public service that would be ok to use for this purpose? If not, then I think using a static-server service is better.